File: JavaScript/6js 3/toNato.js
/*
|--------------------------------------------------------------------------
| If you can read this...
|--------------------------------------------------------------------------
|
*/

function toNato(words) {
   const NATO = {
      A: "Alfa",
      B: "Bravo",
      C: "Charlie",
      D: "Delta",
      E: "Echo",
      F: "Foxtrot",
      G: "Golf",
      H: "Hotel",
      I: "India",
      J: "Juliett",
      K: "Kilo",
      L: "Lima",
      M: "Mike",
      N: "November",
      O: "Oscar",
      P: "Papa",
      Q: "Quebec",
      R: "Romeo",
      S: "Sierra",
      T: "Tango",
      U: "Uniform",
      V: "Victor",
      W: "Whiskey",
      X: "Xray",
      Y: "Yankee",
      Z: "Zulu",
   };
   let res = [];
   for (let char of words) {
      if (char.match(/[a-zA-Z]/)) {
         res.push(NATO[char.toUpperCase()]);
      } else if (char.match(/[.,!?]/)) {
         res.push(char);
      }
   }
   return res.join(" ");
}


const toNato = (w) => w.replace(/\w ?/g, (m) => `${NATO[m[0].toUpperCase()]} ` || m).trim();


function to_nato(words) {
   return words
      .split("")
      .filter((c) => c !== " ")
      .map((c) => table[c.toUpperCase()] || c)
      .join(" ");
}


console.log(
   toNato("If, you can read?"),
   "India Foxtrot , Yankee Oscar Uniform Charlie Alfa November Romeo Echo Alfa Delta ?"
);

File: Python/6py 3/to_nato.py
"""
|--------------------------------------------------------------------------
| If you can read this...
|--------------------------------------------------------------------------
|
"""

from preloaded import NATO
def to_nato(words):
   res = []
   for char in words:
      if char.isalpha():
         res.append(NATO[char.upper()])
      elif char in '.,!?':
         res.append(char)
   return ' '.join(res)


def to_nato(words):
   return " ".join(NATO.get(c, c) for c in words.upper() if c != " ")


def to_nato(words):
   return ' '.join(NATO.get(i,i) for i in words.upper().replace(' ',''))


print(to_nato('If you can read'), "India Foxtrot Yankee Oscar Uniform Charlie Alfa November Romeo Echo Alfa Delta")
print(to_nato('Did not see that coming'), "Delta India Delta November Oscar Tango Sierra Echo Echo Tango Hotel Alfa Tango Charlie Oscar Mike India November Golf")
print(to_nato('.d?d!'),'. Delta ? Delta !')